src/components/calendar/calendar.jsx

```diff
@@ -0,0 +1,199 @@
+import React, {
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import arrowLeft from '../../svgs/arrow-left.svg';
+import arrowRight from '../../svgs/arrow-right.svg';
+import caretDown from '../../svgs/caret-down.svg';
+import IconButton from '../icon-button/icon-button.jsx';
+import Icon from '../icon/icon.jsx';
+import styles from './calendar.css';
+
+function getDateIterator(year, month) {
+  const firstDate = new Date(year, month, 1);
+  return new Date(year, month, 1 - firstDate.getDay());
+}
+
+function getHeadCell(iterator) {
+  const date = iterator.getDate();
+  const weekday = iterator.toLocaleDateString(undefined, {
+    weekday: 'narrow',
+  });
+
+  // This needs to occur at the end of this function
+  iterator.setDate(date + 1);
+
+  return (
+    <th className={styles.weekday}>
+      {weekday}
+    </th>
+  );
+}
+
+function getCellClassName(iterator, month, highlightedDate) {
+  if (
```

Comment:
Could we extract some of these out? How about

```suggestion
const matchesYear = highlightedDate.getFullYear() === iterator.getFullYear();
const matchesMonth = highlightedDate.getMonth() === iterator.getMonth();
const matchesDate = highlightedDate.getDate() === iterator.getDate();

if (matchesYear && matchesMonth && matchesDate) {

```

Reply:
I hoisted something similar. I'm still not sure the value it provides.